app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,18 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time, threshold=1.day)
+    raise _("Start time must be earlier than End time") if start_time > end_time
+    # Create an array of ordered pairs from start_time and end_time so that each ordered pair is contained
+    # within the threshold.  Then, reverse it so the newest ordered pair is first:
+    # start_time = 2017/01/01 12:00:00, end_time = 2017/01/04 12:00:00
+    # [[2017-01-03 12:00:00 UTC, 2017-01-04 12:00:00 UTC],
+    #  [2017-01-02 12:00:00 UTC, 2017-01-03 12:00:00 UTC],
+    #  [2017-01-01 12:00:00 UTC, 2017-01-02 12:00:00 UTC]]
+    (start_time..end_time).step_value(threshold).each_cons(2).collect { |s_time, e_time| [s_time, e_time] }.reverse
```

Comment:
To your point though, if you pass in interval_name, then you just call map directly on each_cons.

```ruby
> (1..20).step_value(2).each_cons(2).map { |t1, t2| [interval_name, t1, t2] }.reverse
[5] pry(main)> (1..20).step_value(2).each_cons(2).map { |t1, t2| [interval_name, t1, t2] }.reverse
=> [["realtime", 19, 20],
 ["realtime", 17, 19],
 ["realtime", 15, 17],
 ["realtime", 13, 15],
 ["realtime", 11, 13],
 ["realtime", 9, 11],
 ["realtime", 7, 9],
 ["realtime", 5, 7],
 ["realtime", 3, 5],
 ["realtime", 1, 3]]
```